qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -112,10 +119,18 @@ def __init__(self, data, coeffs=None, *, ignore_pauli_phase=False, copy=True):
 
         pauli_list = PauliList(data.copy() if copy and hasattr(data, "copy") else data)
 
-        dtype = object if isinstance(coeffs, np.ndarray) and coeffs.dtype == object else complex
+        if isinstance(coeffs, np.ndarray) and coeffs.dtype == object:
+            dtype = object
+        elif coeffs is not None:
+            if not isinstance(coeffs, (np.ndarray, Sequence)):
+                coeffs = [coeffs]
+            if any(isinstance(coeff, ParameterExpression) for coeff in coeffs):
+                dtype = object
+            else:
+                dtype = complex
 
         if coeffs is None:
-            coeffs = np.ones(pauli_list.size, dtype=dtype)
+            coeffs = np.ones(pauli_list.size, dtype=complex)
```

Comment:
Why not keep `dtype=dtype`? It should have been assigned above, right?

Reply:
If `coeffs` is none, we cannot determine the type of coeffs, so the default should be complex.